### PR TITLE
Add a stresstest for verifying memory not changing under your feet.

### PR DESCRIPTION
**Pull request by baldersheim**
Both verify heap memory, and stress file backed swap like memory.

@vekterli PR

### CODE REVIEW

**Review by vekterli — APPROVED** [review]
:+1:

File: vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    {
+        std::lock_guard guard(_mutex);
+        alloc = _allocations[std::rand() % _allocations.size()].get();
+    }
+    size_t error_count = 0;
+    for (size_t i = 0; i < cfg().alloc_size(); i++) {
+        if (alloc[i] != 0) {
+            error_count++;
+            std::lock_guard guard(log_mutex);
+            std::cout << "Tread " << std::this_thread::get_id() << ": Unexpected byte(" << std::hex << int(alloc[i]) << ") at " << (alloc + i) << std::endl;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp:110]: Fixed

[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp:110]: Fixed

File: vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    std::string swap_dir;
+    std::cout << description << std::endl;
+    for (int i = 1; i+2 <= argc; i+=2) {
+        char option = argv[i][strlen(argv[i]) - 1];
+        char *arg = argv[i+1];
+        switch (option) {
+            case 'h': heapSize = atof(arg) * 1_Gi;
+            break;
+            case 's': swap_dir = arg;
+            break;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp:222]: Fixed

File: vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+}
+
+int
+main(int argc, char *argv[]) {
+    size_t heapSize = 1_Gi;
+    duration runTime = 10s;
+    std::string swap_dir;
+    std::cout << description << std::endl;
+    for (int i = 1; i+2 <= argc; i+=2) {
+        char option = argv[i][strlen(argv[i]) - 1];
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp:216]: Leaving as is for now

File: vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    _allocations.push_back(std::move(alloc));
+    return 1;
+}
+
+size_t
+Allocations::verify_random_allocation() const {
+    const char * alloc;
+    {
+        std::lock_guard guard(_mutex);
+        alloc = _allocations[std::rand() % _allocations.size()].get();
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespalib/src/apps/vespa-stress-and-validate-memory/stress_and_validate_memory.cpp:95]: rand_r

**Pull request merged**